AI troops assignment logic.

SpiritBaker opened the pull request:
AI will operate from his borders or allied borders during the war.
AI armies will be more concentrated in area with a wargoal.

SpiritBaker committed:
AI troops assignment logic.

stackpoint commented on common/static_modifiers/war_taxes.txt:
```diff
@@ -0,0 +1,5 @@
+	land_maintenance_modifier = -0.15
+	naval_maintenance_modifier = -0.15
+	reinforce_cost_modifier = -0.2
+}
```
Does the AI use war taxes?

SpiritBaker replied:
yes it does

SpiritBaker committed:
Fixes to docks/galley ideas

Making docks line up available without idea investment. It's kinda sad that all Navy building require Naval ideas.
Adjusting everything connected to that.

stackpoint committed:
Merge branch 'master' into master
